src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt

```diff
@@ -70,28 +72,59 @@ data class UserPersonDetail(
 
   override fun hashCode(): Int = username.hashCode()
 
+  private fun findCaseloadById(caseloadId: String): Caseload? {
+    return caseloads.firstOrNull { caseload -> caseloadId == caseload.id.caseloadId }?.caseload
+  }
+
+  fun setDefaultCaseload(caseloadId: String) {
+    activeCaseLoad = findCaseloadById(caseloadId) ?: throw CaseloadNotFoundException("Default caseload cannot be set as user does not have $caseloadId.")
+  }
+
   fun addCaseload(caseload: Caseload) {
+    findCaseloadById(caseloadId = caseload.id)?.let { throw CaseloadAlreadyExistsException("Caseload ${it.id} already added to this user") }
+
```

Comment:
very very minor but semantically `.run` is better than `let` since are not returning a result